src/api.rs

```diff
@@ -203,8 +201,8 @@ unsafe fn join_inject<A,B,RA,RB>(oper_a: A,
 
     thread_pool::get_registry().inject(&[job_a.as_job_ref(), job_b.as_job_ref()]);
 
-    job_a.latch.wait();
-    job_b.latch.wait();
+    job_a.wait();
```

Comment:
Why did the `wait()` method move onto the `Task`? Is this just general cleanup or is it needed for some other reason?